common/src/main/java/com/wynntils/models/content/ContentModel.java

```diff
@@ -0,0 +1,160 @@
+    private static final Pattern DIFFICULTY_PATTERN = Pattern.compile("^   §7Difficulty: (\\w*)$");
+    private static final Pattern REWARD_HEADER_PATTERN = Pattern.compile("^   §dRewards:$");
+    private static final Pattern REWARD_PATTERN = Pattern.compile("^   §d- §7\\+?(.*)$");
+    private static final Pattern TRACKING_PATTERN = Pattern.compile("^ *À*§.§lCLICK TO (UN)?TRACK$");
+
+    public ContentModel() {
+        super(List.of());
+    }
+
+    public ContentInfo parseItem(String name, ContentType type, ItemStack itemStack) {
+        LinkedList<StyledText> lore = LoreUtils.getLore(itemStack);
+
+        String statusLine = lore.pop().getString();
+        if (!statusLine.substring(0, 1).equals("§")) return null;
+
+        ContentStatus status = ContentStatus.from(statusLine.substring(1, 2), itemStack.getItem());
+        int specialInfoEnd = statusLine.indexOf(" - ");
+        // If we have a specialInfo, skip the §x marker in the beginning, and keep everything
+        // until the " - " comes.
+        String specialInfo = specialInfoEnd != -1 ? statusLine.substring(2, specialInfoEnd) : null;
+        if (!lore.pop().isEmpty()) return null;
+
+        Pair<Integer, Boolean> levelReq = Pair.of(0, true);
+        ContentDistance distance = null;
+        String distanceInfo = null;
+        ContentLength length = null;
+        String lengthInfo = null;
+        ContentDifficulty difficulty = null;
+        ContentTrackingState trackingState = ContentTrackingState.UNTRACKABLE;
+        List<Pair<Pair<ProfessionType, Integer>, Boolean>> professionLevels = new ArrayList<>();
+        List<Pair<String, Boolean>> quests = new ArrayList<>();
+        List<String> rewards = new ArrayList<>();
+        List<StyledText> descriptionLines = new ArrayList<>();
+
+        for (StyledText line : lore) {
+            // Must be tested before profession requirement pattern
+            Matcher levelReqMatcher = line.getMatcher(LEVEL_REQ_PATTERN);
+            if (levelReqMatcher.matches()) {
+                boolean fulfilled = levelReqMatcher.group(1).equals("a");
+                int level = Integer.parseInt(levelReqMatcher.group(2));
+                levelReq = Pair.of(level, fulfilled);
+                continue;
+            }
+
+            Matcher professionReqMatcher = line.getMatcher(PROFESSION_REQ_PATTERN);
+            if (professionReqMatcher.matches()) {
+                boolean fulfilled = professionReqMatcher.group(1).equals("a");
+                ProfessionType profession = ProfessionType.fromString(professionReqMatcher.group(2));
+                int level = Integer.parseInt(professionReqMatcher.group(3));
+                professionLevels.add(Pair.of(Pair.of(profession, level), fulfilled));
+                continue;
+            }
+
+            Matcher questReqMatcher = line.getMatcher(QUEST_REQ_PATTERN);
+            if (questReqMatcher.matches()) {
+                boolean fulfilled = questReqMatcher.group(1).equals("a");
```

Comment:
This equals here (and above) is admittedly weird. I know that you check for the GREEN color, because I worked enough with ChatFormatting to memorize most codes (thanks StyledText). But for anyone else, this is the weirdest check ever. Perhaps change "a" to ChatFormatting.GREEN.code() or whatever the method is called.

Reply:
I refactored this so it is hopefully more comprehensible.